libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/internal/MapboxNavigationViewApi.kt

```diff
@@ -0,0 +1,105 @@
+package com.mapbox.navigation.dropin.internal
+
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.dropin.NavigationViewApi
+import com.mapbox.navigation.ui.app.internal.Store
+import com.mapbox.navigation.ui.app.internal.destination.Destination
+import com.mapbox.navigation.ui.app.internal.destination.DestinationAction
+import com.mapbox.navigation.ui.app.internal.extension.dispatch
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationStateAction
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewAction
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewState
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
+import com.mapbox.navigation.ui.app.internal.tripsession.TripSessionStarterAction
+import com.mapbox.navigation.ui.app.internal.endNavigation as endNavigationAction
+
+@ExperimentalPreviewMapboxNavigationAPI
+internal class MapboxNavigationViewApi(
+    private val store: Store
+) : NavigationViewApi() {
+
+    override fun fetchRoutes(options: RouteOptions) {
+        store.dispatch(RoutePreviewAction.FetchOptions(options))
+    }
+
+    override fun fetchRoutes(points: List<Point>) {
+        store.dispatch(RoutePreviewAction.FetchPoints(points))
+    }
+
+    override fun setPreviewRoutes(routes: List<NavigationRoute>) {
+        store.dispatch(RoutePreviewAction.Ready(routes))
+    }
+
+    override fun setRoutes(routes: List<NavigationRoute>) {
+        store.dispatch(RoutesAction.SetRoutes(routes))
+    }
+
+    override fun setDestination(point: Point) {
+        store.dispatch(DestinationAction.SetDestination(Destination(point)))
+    }
+
+    override fun enableTripSession() {
+        store.dispatch(TripSessionStarterAction.EnableTripSession)
+    }
+
+    override fun enableReplaySession() {
+        store.dispatch(TripSessionStarterAction.EnableReplayTripSession)
+    }
+
+    override fun isReplayEnabled(): Boolean {
+        return store.state.value.tripSession.isReplayEnabled
+    }
+
+    override fun startFreeDrive() {
+        store.dispatch(RoutesAction.SetRoutes(emptyList()))
+        store.dispatch(RoutePreviewAction.Ready(emptyList()))
+        store.dispatch(NavigationStateAction.Update(NavigationState.FreeDrive))
```

Comment:
Should this be needed? It seems the `NavigationStateController` should be able to figure out that it is in `FreeDrive` when there are no routes and the trip session is started.

Otherwise, this `startFreeDrive` appears to be a required function in order to put the `NavigationView` into free drive 🤔

Reply:
🤷 
remember [this](https://github.com/mapbox/mapbox-navigation-android/blob/e4fab080d536984928e60cc7f4c99ab6526fbf09/libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/navigation/NavigationStateComponent.kt)?

Reply:
> Should this be needed? It seems the `NavigationStateController` should be able to figure out that it is in `FreeDrive` when there are no routes and the trip session is started.
> 
> Otherwise, this `startFreeDrive` appears to be a required function in order to put the `NavigationView` into free drive 🤔 

@kmadsen If an end user has a different use case and they want Drop-in to enter free drive state via different UI control, how would they do that if we don't have this function? Moreover, what you are saying is true in the case of the Core SDK. In Drop-In it's not sufficient to just set routes to null with trip session enabled to tell the user you are in free drive. You would need to update the `NavigationState` to free drive to control the UI. So for Drop-In, if you were to press on some custom external button in active navigation mode you need to tell Drop-In to set routes to null and transition to free drive. That's where this API comes in.

Reply:
I'm mostly skeptic of having _any logic_ inside this api layer.

Because where are the unit tests? Sure the function `startFreeDrive` should exist. But it seems it should be in a logic layer. For example, why isn't there a `StartFreeDriveAction`?

Reply:
I would like to have the logic outside API as well. Hence, I put up this [comment](https://github.com/mapbox/mapbox-navigation-android/pull/6174#discussion_r950639061) as well.

I would not block this PR though as it introduces the necessary public API(s) and rest all is implementation details which can be changed later.